### ISSUE TITLE
donor_organism.organism_age example doesn't match pattern

### ISSUE THREAD

**Issue by hannes-ucsc** [HumanCellAtlas/metadata-schema#1320]
https://github.com/HumanCellAtlas/metadata-schema/blob/c2ab8a5115a2fabd4fcbb57fda89022be12cd939/json_schema/type/biomaterial/donor_organism.json#L96

It appears that sInce draft 6, one can specify multiple example values: 

https://json-schema.org/understanding-json-schema/reference/generic.html

It would be prudent to validate the schemas for this type of problem.

**Comment by mshadbolt** [comment]
So I think what you are saying is that the full string: `"20; 45-65"` 
does not match the regex: `"^[0-9]+\\.?[0-9]*-?[0-9]*\\.?[0-9]*$"`

While this is true, in our schema, separate examples are delimited by a `; `, i.e. here there are two examples, `20` and `45-65`. When we run the schema linter before we merge a PR, it is able to separate these and check that each example is valid against the schema. (https://github.com/HumanCellAtlas/metadata-schema/blob/c2ab8a5115a2fabd4fcbb57fda89022be12cd939/src/schema_linter.py#L205)

I wasn't around when the schemas were originally written so not sure why this approach was taken. 

I also wasn't aware that you could instead specify examples as an array but agree this might have been a more robust approach.

We could look into transitioning to an array of examples in future.

**Comment by hannes-ucsc** [comment]
https://json-schema.org/understanding-json-schema/reference/generic.html

> Each entry should validate against the schema in which is resides, but that isn’t strictly required.

So I guess the standard recommends but does not mandate it.

> examples are delimited by a ; , i.e. here there are two examples, 20 and 45-65

Is that obvious to users? Where is that convention documented?

**Comment by mshadbolt** [comment]
I am not sure if it is obvious to users, given you are one of our most experienced users and it wasn't obvious to you I would say perhaps not.

For users that use the schema directly it is noted in our schema style guide here: https://github.com/HumanCellAtlas/metadata-schema/blob/master/docs/schema_style_guide.md

For contributors who use our spreadsheet, the examples are populated into a row like this:

![Screenshot 2020-06-30 at 08 03 36](https://user-images.githubusercontent.com/13007987/86094318-41d56e80-baa8-11ea-8326-0e192dcc9efc.png)

I do think contributors generally recognise that these are two different examples. I think where contributors do get confused is for fields that they can specify multiple values, they interpret the example to mean that they should also separate values in the field with a `;` whereas we expect multiple values to be separated with `||`

For example one is able to specify multiple diseases. If a donor has both diabetes and hypertension, following the example text they might put
```
diabetes; hypertension
```
rather than
```
diabetes||hypertension
```
This would generally be caught during wrangler review.

**Comment by hannes-ucsc** [comment]
Makes sense. I think moving from `example` to `examples` and some special logic for rendering the spreadsheet header could be a possible solution.

So for example, `"examples": [ "one", "two", "three" ]` in the schema could be rendered in the spreadsheet header row as `For example: "one", "two" or "three"`.

But given that the schema spec doesn't require `example` to be valid, this isn't as urgent as I thought. It's not a defect, only an opportunity to improve user experience.
